src/Utils/Snapping.vala

```diff
@@ -0,0 +1,296 @@
+    ) {
+        var grid = SnapGrid ();
+        grid.v_snaps = new Gee.HashMap<int, SnapMeta> ();
+        grid.h_snaps = new Gee.HashMap<int, SnapMeta> ();
+
+        foreach (var cand in v_candidates) {
+          var candidate_item = cand as Lib.Items.CanvasItem;
+          if (candidate_item != null && selection.find (candidate_item) == null) {
+            populate_vertical_snaps (candidate_item, ref grid.v_snaps);
+          }
+        }
+
+        foreach (var cand in h_candidates) {
+          var candidate_item = cand as Lib.Items.CanvasItem;
+          if (candidate_item != null && selection.find (candidate_item) == null) {
+            populate_horizontal_snaps (candidate_item, ref grid.h_snaps);
+          }
+        }
+
+        return grid;
+    }
+
+    /// Populates the horizontal snaps of an item
+    private static void populate_horizontal_snaps (Lib.Items.CanvasItem item, ref Gee.HashMap<int, SnapMeta> map) {
+        int x_1 = (int)item.bounds.x1;
+        int x_2 = (int)item.bounds.x2;
+        int y_1 = (int)item.bounds.y1;
+        int y_2 = (int)item.bounds.y2;
+        int center_x = (int)(Math.ceil((item.bounds.x2 - item.bounds.x1) / 2.0) + item.bounds.x1);
+        int center_y = (int)(Math.ceil((item.bounds.y2 - item.bounds.y1) / 2.0) + item.bounds.y1);
+
+        add_to_map (x_1, y_1, y_2, center_y, -1, ref map);
+        add_to_map (x_2, y_1, y_2, center_y, 1, ref map);
+        add_to_map (center_x, center_y, center_y, center_y, 0, ref map);
+    }
+
+    /// Populates the vertical snaps of an item
+    private static void populate_vertical_snaps (Lib.Items.CanvasItem item, ref Gee.HashMap<int, SnapMeta> map) {
+        int x_1 = (int)item.bounds.x1;
+        int x_2 = (int)item.bounds.x2;
+        int y_1 = (int)item.bounds.y1;
+        int y_2 = (int)item.bounds.y2;
+        int center_x = (int)(Math.ceil((item.bounds.x2 - item.bounds.x1) / 2.0) + item.bounds.x1);
+        int center_y = (int)(Math.ceil((item.bounds.y2 - item.bounds.y1) / 2.0) + item.bounds.y1);
+
+        add_to_map (y_1, x_1, x_2, center_x, -1, ref map);
+        add_to_map (y_2, x_1, x_2, center_x, 1, ref map);
+        add_to_map (center_y, center_x, center_x, center_x, 0, ref map);
+    }
+
+    /// Simple method to add information to a snap list
+    private static void add_to_map (
+        int pos,
+        int n1,
+        int n2,
+        int n3,
+        int polarity,
+        ref Gee.HashMap<int, Utils.Snapping.SnapMeta> map
+    ) {
+        if (map.has_key (pos)) {
+            SnapMeta k = map.get (pos);
+            k.normals.add (n1);
+            k.normals.add (n2);
+            k.normals.add (n3);
+            k.polarity += polarity;
+        }
+        else {
+            var v = new SnapMeta ();
+            v.normals = new Gee.HashSet<int> ();
+            v.normals.add (n1);
+            v.normals.add (n2);
+            v.normals.add (n3);
+            v.polarity = polarity;
+            map.set (pos, v);
+        }
```

Comment:
Since nothing happens after the callback condition, we can do an early `return;` in the first condition, and avoid the else.